packages/gatsby-theme-docs/src/hooks/use-active-section.js

```diff
@@ -15,16 +15,32 @@ const getSectionElements = () =>
 
 const offset = calculateOffset();
 
+const isInViewport = (element) => {
+  const rect = element.getBoundingClientRect();
+
+  return (
```

Comment:
This has to be fine tuned and properly tested, it's here just to give an idea

Reply:
I think a comment verbalizing what the expression tests in layout terminology would help maintenance  - and me being able to review it ;-) .

Testing the console output seems to yield the correct output (at least without more intensive testing e.g. with scrollbars when no touch mouse is attached and the laptop is closed)